src/tools/rbd_mirror/image_replayer/snapshot/Replayer.cc

```diff
@@ -1563,17 +1576,36 @@ void Replayer<I>::register_perf_counters() {
 
   auto cct = static_cast<CephContext *>(m_state_builder->local_image_ctx->cct);
   auto prio = cct->_conf.get_val<int64_t>("rbd_mirror_image_perf_stats_prio");
-  PerfCountersBuilder plb(g_ceph_context,
-                          "rbd_mirror_snapshot_image_" + m_image_spec,
-                          l_rbd_mirror_snapshot_first,
+
+  auto *local_image_ctx = m_state_builder->local_image_ctx;
+  std::string labels = ceph::perf_counters::key_create(
+      "rbd_mirror_snapshot",
+      {{"pool", local_image_ctx->md_ctx.get_pool_name()},
+       {"namespace", local_image_ctx->md_ctx.get_namespace()},
+       {"image", local_image_ctx->name}});
+
+  PerfCountersBuilder plb(g_ceph_context, labels, l_rbd_mirror_snapshot_first,
                           l_rbd_mirror_snapshot_last);
-  plb.add_u64_counter(l_rbd_mirror_snapshot_replay_snapshots,
-                      "snapshots", "Snapshots", "r", prio);
+  plb.add_u64_counter(l_rbd_mirror_snapshot_replay_snapshots, "snapshots",
+                      "Snapshots", "r", prio);
   plb.add_time_avg(l_rbd_mirror_snapshot_replay_snapshots_time,
                    "snapshots_time", "Snapshots time", "rl", prio);
   plb.add_u64_counter(l_rbd_mirror_snapshot_replay_bytes, "replay_bytes",
                       "Replayed data", "rb", prio, unit_t(UNIT_BYTES));
+  plb.add_u64_counter(l_rbd_mirror_snapshot_remote_timestamp,
+                      "remote_timestamp", "Timestamp of the remote snapshot",
+                      "rt", prio);
+  plb.add_u64_counter(l_rbd_mirror_snapshot_local_timestamp, "local_timestamp",
+                      "Timestamp of the local snapshot", "lt", prio);
+  plb.add_u64_counter(l_rbd_mirror_snapshot_last_sync_seconds,
+                      "last_sync_seconds",
+                      "Time taken to sync the last snapshot", "st", prio);
+  plb.add_u64_counter(l_rbd_mirror_snapshot_last_sync_bytes, "last_sync_bytes",
+                      "Bytes synced for the last snapshot", "sb", prio);
```

Comment:
Overall, I see a bunch of things wrong here:
- `last_sync_bytes` can go up and down, so it should be a gauge, not a counter.  It's also missing a units designator.
- Timestamps should have the proper type ("time"), not pretend to be a counter.
- Counter nicks don't make much sense, especially for existing counters that were cut-and-pasted from journal-based mirroring code ("Snapshots" vs "r", "Snapshots time" vs "rl", etc).  I think we should get rid of them entirely.
- Having the word "replay" in snapshot-based mirroring counter names can be confusing since it's the journal that is replayed.  Snapshots are synced -- at least that is the term used elsewhere.

Since the move to labeled counters breaks backwards compatibility anyway, let's use this opportunity to clean all of this up.  In the interest of time, I'm pasting proposed perf counter definitions directly.  Please review and, if no objections, incorporate along with the necessary fixups in places where they are used:
```
  plb.add_u64_counter(l_rbd_mirror_snapshot_snapshots, "snapshots",
                      "Number of snapshots synced", nullptr, prio);
  plb.add_time_avg(l_rbd_mirror_snapshot_sync_time, "sync_time",
                   "Average sync time", nullptr, prio);
  plb.add_u64_counter(l_rbd_mirror_snapshot_sync_bytes, "sync_bytes",
                      "Total bytes synced", nullptr, prio, unit_t(UNIT_BYTES));
  plb.add_time(l_rbd_mirror_snapshot_remote_timestamp, "remote_timestamp",
               "Timestamp of the remote snapshot", nullptr, prio);
  plb.add_time(l_rbd_mirror_snapshot_local_timestamp, "local_timestamp",
               "Timestamp of the local snapshot", nullptr, prio);
  plb.add_time(l_rbd_mirror_snapshot_last_sync_time, "last_sync_time",
               "Time taken to sync the last snapshot", nullptr, prio);
  plb.add_u64(l_rbd_mirror_snapshot_last_sync_bytes, "last_sync_bytes",
              "Bytes synced for the last snapshot", nullptr, prio,
              unit_t(UNIT_BYTES));
```